src/main/java/htsjdk/samtools/CachingBAMFileIndexOptimized.java

```diff
@@ -0,0 +1,69 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.seekablestream.SeekableStream;
+
+/**
+ * A subclass of CachingBAMFileIndex that is optimized for merging, by returning
+ * null BAMIndexContent objects if all bins are empty.
+ */
+public class CachingBAMFileIndexOptimized extends CachingBAMFileIndex {
```

Comment:
Renamed.

Reply:
I tried moving the mergers into one package but it would mean that we have to make lots of classes/methods on `Bin`, `BAMIndexMetadata`, `BAMIndexContent` public, which doesn't seem particularly desirable. Also, it breaks the pattern in htsjdk where packages are used to group formats rather than operations, e.g. a `cram` package and a `variant` package - which goes against having a `merger` package.